medusa/providers/torrent/html/shanaproject.py

```diff
@@ -0,0 +1,190 @@
+# coding=utf-8
+
+"""Provider code for ShanaProject."""
+
+from __future__ import unicode_literals
+
+import logging
+import re
+import traceback
+
+from dateutil import parser
+
+from medusa import tv
+from medusa.bs4_parser import BS4Parser
+from medusa.helper.common import convert_size
+from medusa.logger.adapters.style import BraceAdapter
+from medusa.providers.torrent.torrent_provider import TorrentProvider
+
+from requests.compat import urljoin
+
+log = BraceAdapter(logging.getLogger(__name__))
+log.logger.addHandler(logging.NullHandler())
+
+
+class ShanaProjectProvider(TorrentProvider):
+    """ShanaProject Torrent provider."""
+
+    size_regex = re.compile('([\d.]+)(.*)')
+
+    def __init__(self):
+        """Initialize the class."""
+        super(ShanaProjectProvider, self).__init__('ShanaProject')
+
+        # Credentials
+        self.public = True
+
+        # URLs
+        self.url = 'https://www.shanaproject.com/'
+        self.urls = {
+            'search': urljoin(self.url, '/search/{page}/'),
+        }
+
+        # Miscellaneous Options
+        self.max_pages = 3  # Max return 150 results for 3 pages.
+        self.supports_absolute_numbering = True
+        self.anime_only = True
+
+        # Torrent Stats
+        self.minseed = None
+        self.minleech = None
+
+        # Cache
+        self.cache = tv.Cache(self, min_time=20)
+
+    def search(self, search_strings, age=0, ep_obj=None):
```

Comment:
Yes.  If the params were called by keyword instead of positionally it would break.